File: swift/69_SqrtX.swift
class Solution {
    func mySqrt(_ x: Int) -> Int {
        if x == 1 { return 1 }
        var left = 1, right = x / 2
        while left <= right {
            var mid = left + (right - left) / 2
            var squared = mid * mid
            if squared == x { return mid }
            else if squared > x { right = mid - 1 }
            else { left = mid + 1 }
        }
        return right
    }
}

class Solution {
    func mySqrt(_ x: Int) -> Int {
        return Int(sqrt(Double(x)))
    }
}

File: python/69_SqrtX.py
class Solution:
    def mySqrt(self, x: int) -> int:
        if x == 1: return 1
        left, right = 1, x // 2
        while left <= right:
            mid = left + (right - left) // 2
            sq = mid * mid
            if sq == x: return mid
            elif sq > x: right = mid - 1
            else: left = mid + 1
        return right

class Solution:
    def mySqrt(self, x: int) -> int:
        return int(x**.5)
